### DISCUSSION TITLE
lamellar

### DISCUSSION THREAD

**Discussion by kittygg** [neperfepx/neper#558] [Q&A] [answered]
How to generate different numbers of grains in different layers. see the schematic. Thank you!
![p](https://user-images.githubusercontent.com/110077741/189871748-a4cf4eb3-34eb-4c08-9acc-89b54a71581a.jpg)

**Comment by rquey** [comment, marked as answer]
Use a multiscale cell file:

`scale2`:
```
1 5
2 10
3 15
4 15
5 8
```

```
neper -T -n "from_morpho::msfile(scale2)" -morpho "lamellar(w=0.27,pos=start,v=(1,2,0))::voronoi" -dim 2 -o d
neper -V d.tess -datacellcol "scaleid(1)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/189914684-dd4a7137-26b6-4d46-aeec-e0e65c317c4c.png)

**Reply by kittygg** [reply]
Thanks!  How to make the grains in each layer obey different distributions? such as normal, lognormal.

**Reply by rquey** [reply]
Use a multiscale cell file, but for `-morpho`:

`scale2b`:
```
1 diameq:normal(1,0.1),1-sphericity:lognormal(0.15,0.03)
2 diameq:normal(1,0.2),1-sphericity:lognormal(0.15,0.03)
3 diameq:lognormal(1,0.3),1-sphericity:lognormal(0.15,0.03)
4 diameq:lognormal(1,0.5),1-sphericity:lognormal(0.15,0.03)
5 diameq:lognormal(1,0.7),1-sphericity:lognormal(0.15,0.03)
```

```
neper -T -n "from_morpho::msfile(scale2)" -morpho "lamellar(w=0.27,pos=start,v=(1,2,0))::msfile(scale2b)" -dim 2 -o d
neper -V d.tess -datacellcol "scaleid(1)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/190085256-6f96a6f0-f5d7-4f5a-bffe-76274982d0fe.png)
